tests/functional/b/bad_thread_instantiation.py

```diff
@@ -1,8 +1,10 @@
-# pylint: disable=missing-docstring
+# pylint: disable=missing-docstring, redundant-keyword-arg
 import threading
 
 
 threading.Thread(lambda: None).run()  # [bad-thread-instantiation]
-threading.Thread(None, lambda: None)
+threading.Thread(None, lambda: None)  # [bad-thread-instantiation]
```

Comment:
This wasn't raising previously, probably voluntarily as the test existed and there's no comment around it, did you check the reasoning in the original implementation?

Reply:
The reason this now raises is 💯 to do with this new PR. If any call to thread.Thread does NOT use `target=` as a kwarg, it should raise the warning, so this change makes perfect sense.  Previous impl didn't desire to do that, so it doesn't mention anything

Reply:
I don't think this is a good change and might annoy users. Arguments are parsed in order so supplying `target` as the second argument is perfectly valid. Why would we change that?

_This is similar as requiring `maxsplit` in `str.split()` to always be passed as a kwarg. It seems unnecessarily pedantic._

Reply:
This change definitely dances the line of false positive/negatives. The difference in `threading.Thread` v. something like `split` is how likely we think users are to confuse the first and second arg. As mentioned in the current docs `By default, the first parameter is the group param, not the target param.`. So users tend to think that the first argument is the target func, but it isn't.

Happy to hear counter test cases which would solve the original false negative posed in the issue
`thread = threading.Thread(thread_target, args=(10,))`

Reply:
Why can't we catch the example from the issue but also accept `target` without they keyword?

Reply:
I'm not sure how we would catch the example from the issue otherwise??

Also! Just re-read the [docs](https://docs.python.org/3/library/threading.html#threading.Thread) check this out:
`This constructor should always be called with keyword arguments. Arguments are:`

seems to me that actually gives pylint a strong reason to warn users if they don't use kwargs

Reply:
hm, that is indeed interesting. But then shouldn't we warn for other arguments as well?

Reply:
Sure! Reading that now, it would seem to be appropriate. Want me to add it to this PR? Separate issue to discuss?

Reply:
Yeah I'm not sure. Changing the meaning of the warning seems to be suboptimal.

Reply:
@Pierre-Sassoulas do you have an opinion about this?

Reply:
Sorry I don't have an opinion right now, I'd need to educate myself about threading first :)

Reply:
The comment in the python docs is very old and seems to be applicable to all kinds of methods, not just the case at hand. So like @DanielNoord I'm reluctant to change the meaning of this message to start enforcing passing arguments by keyword, even though it's good practice.

What we should still be able to do in this PR is just check that there is either a `target` keyword arg or a second positional argument.

> I'm not sure how we would catch the example from the issue otherwise??

Hopefully it's trivial, that example had one positional arg, not two, and a kwarg other than `target`.

Reply:
Thanks for the helpful and specific feedback @jacobtylerwalls ! I think I can implement exactly that.